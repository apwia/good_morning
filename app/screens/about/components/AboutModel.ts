type AboutModel = {
  title: string,
  texts: string[],
  urlGitHub: string
};

export default AboutModel