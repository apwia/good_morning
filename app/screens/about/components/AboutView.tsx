import React from 'react';
import { View, Text, FlatList, Image, TouchableOpacity, Linking, } from 'react-native';

import styles from './AboutStyles'
import  useAboutViewModel from './AboutViewModel';
import { SafeAreaView } from 'react-native-safe-area-context';

const AboutView = () => {

  const {title, texts, urlGitHub} = useAboutViewModel();

  return (

    <SafeAreaView>
      <View style={styles.container}>

        <Text style={styles.title}>{title}</Text>

        <FlatList 
          data={texts} 
          renderItem={({item}) => <Text style={styles.description}>{item}</Text> }
          keyExtractor={item => item.toString()}
        />

        <TouchableOpacity onPress={() => Linking.openURL(urlGitHub)} testID='link'>
          <Image 
            style={styles.image}
            source={require('../../../assets/images/GitHub.png')}/>
        </TouchableOpacity>

      </View>
    </SafeAreaView>

  );
};

export default AboutView;
