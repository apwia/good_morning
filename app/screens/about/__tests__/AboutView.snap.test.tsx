import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import AboutView from '../components/AboutView';
import AboutModel from '../components/AboutModel';


jest.mock('../components/AboutViewModel', () => {
  const fnAboutViewModel = jest.fn();
  fnAboutViewModel.mockImplementation(
    () => 
      ({
        title: "test title",
        texts: ["text1", "text2", "text3"],
        urlGitHub: "testUrl.com"
       } as AboutModel),
  );

  return fnAboutViewModel;
  
});

import useAboutViewModel from '../components/AboutViewModel';
import { fireEvent, render, screen } from '@testing-library/react-native';

it('renders correctly', () => {
  const {toJSON} = renderer.create(<AboutView />);
  expect(toJSON()).toMatchSnapshot();
});

it('should redirect to some page', () => {
  const consoleSpy = jest.spyOn(console, 'log');

  jest.mock('react-native/Libraries/Linking/Linking', () => ({
    openURL: jest.fn((param) => console.log(param)),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  }));

  render(<AboutView/>);

  fireEvent.press(screen.getByTestId('link'));

  expect(consoleSpy).toHaveBeenCalledWith("testUrl.com");

});