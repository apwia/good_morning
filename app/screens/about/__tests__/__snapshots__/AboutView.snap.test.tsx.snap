// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginTop": 30,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#717171",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 30,
        }
      }
    >
      test title
    </Text>
    <RCTScrollView
      data={
        [
          "text1",
          "text2",
          "text3",
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#717171",
                "fontSize": 16,
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            text1
          </Text>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#717171",
                "fontSize": 16,
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            text2
          </Text>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#717171",
                "fontSize": 16,
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            text3
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="link"
    >
      <Image
        source={
          {
            "testUri": "../../../app/assets/images/GitHub.png",
          }
        }
        style={
          {
            "height": 70,
            "width": 80,
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
